apps/web/src/app/[locale]/challenge/_components/sheet-content-custom.tsx

```diff
@@ -0,0 +1,115 @@
+import { SheetHeader, SheetTitle } from '@repo/ui/components/sheet';
+import Link from 'next/link';
+import { ChevronRight } from '@repo/ui/icons';
+import { TrackChallenge } from '../../tracks/_components/track-challenge-card';
+import { TrackProgress } from '../../tracks/_components/track-progress';
+import { SelectDropdown } from './select-dropdown';
+import { useChallengeRouteData } from '~/app/challenge-route-data.hook';
+import { SORT_KEYS, useProblemExplorerContext } from '~/app/problem-explorer.hooks';
+import { useAllChallengesContext } from '~/app/all-challenges.hook';
+import { useLocalStorage } from '~/utils/useLocalStorage';
+
+export function ExplorerPanel() {
+  const [, setTrackNameStorage] = useLocalStorage('trackName', 'popular');
+  const { getTrack, setTrack, title, setTitle, sortKey, setSortKey } = useProblemExplorerContext();
+  const { allChallenges } = useAllChallengesContext();
+
+  const handleValueChange = (value: string) => {
+    if (value === 'popular') {
```

Comment:
This would likely be a good candidate for a switch statement, or perhaps some updates in the logic, such as a map of `value` attributes to the requisite objects which could clean this up a bit.